agcaballero/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -33,54 +33,122 @@
    */ 
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     Collection<String> attendees = request.getAttendees();
-    ArrayList<TimeRange> times = new ArrayList<TimeRange>();
-     // minutes of day is + 1 to account for first and last minute of day
-    boolean[] minutes = new boolean[MINUTES_IN_DAY + 1];
-
-    for(int i = 0; i < minutes.length; i++) {
-      minutes[i] = true;
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    
+    // are there any mandatory employees?
+    boolean mandatory = false;
+    if(attendees.size() > 0) {
+      mandatory = true;
+    }
+    
+    // are there any optional employees?
+    boolean optional = false; 
+    if(optionalAttendees.size() > 0) {
+      optional = true;
     }
 
+    // minutes of day is + 1 to account for first and last minute of day
+    int[] minutes = new int[MINUTES_IN_DAY + 1];
+    // 0 - all attendees (mandatory & optional) can attend 
+    // 1 - mandatory attendees can attend (but not all optional)
+    // 2 - not all mandatory attendees can attend (even if all optional can)
+
     for(Event event: events) {
-      // if there's an overlap in attendees block off those times
-      if(attendeeOverlap(event.getAttendees(), attendees)) {
+      int status = 0;
+      
+      if (attendeeOverlap(event.getAttendees(), attendees)) {
```

Comment:
Suggested name for more grammatical reading: hasOverlappingAttendees()

Reply:
That does sound better. Thank you!

Reply:
I ended up just replacing with the disjoint() function, but this is a better name.

Reply:
Done